controllers/dashboard.go

```diff
@@ -119,7 +110,18 @@ func (c *DashboardController) FileSearch() {
 	flash := web.NewFlash()
 	redirectPath := "/dashboard"
 
-	flash.Success("Scraping all keywords :)")
+	file, fileHeader, err := c.GetFile("file")
+	if err != nil {
+		flash.Error("The specified file could not be uploaded :(")
+	} else {
+		fileForm := form.FileSearchForm{File: file, FileHeader: fileHeader, User: c.CurrentUser}
+		err = fileForm.Save()
+		if err != nil {
+			flash.Error("The specified file could not be uploaded :(")
```

Comment:
The error message needs to be different from the case above for debugging purposes but also for UX purposes. In the first case, there is a problem with the file. In the second case, it could be an error in the application in processing the file.

Also, the CSV upload form has many errors which we should use to show to the user what went wrong e.g. wrong file type, mix/keyword limits

Reply:
I use the `err` returning from the `form` to show the message as `flash` message ✌️